conditional/blueprints/dashboard.py

```diff
@@ -26,117 +19,62 @@
-                      MemberSeminarAttendance.uid == user_name
-                  )]
-    freshman['ts_total'] = len(t_seminars)
-    attendance = [m.name for m in TechnicalSeminar.query.filter(
-        TechnicalSeminar.id.in_(t_seminars)
-    )]
-
-    freshman['ts_list'] = attendance
-
-    h_meetings = [(m.meeting_id, m.attendance_status) for m in
-                  MemberHouseMeetingAttendance.query.filter(
-                      MemberHouseMeetingAttendance.uid == user_name)]
-    freshman['hm_missed'] = len([h for h in h_meetings if h[1] == "Absent"])
-    freshman['social_events'] = freshman_data.social_events
-    freshman['general_comments'] = freshman_data.other_notes
-    freshman['fresh_proj'] = freshman_data.freshman_project
-    freshman['sig_missed'] = freshman_data.signatures_missed
-    freshman['eval_date'] = freshman_data.eval_date
-    return freshman
-
-
-def get_voting_members():
-    voting_list = []
-    active_members = [x['uid'][0].decode('utf-8') for x
-                      in ldap_get_active_members()]
-    intro_members = [x['uid'][0].decode('utf-8') for x
-                     in ldap_get_intro_members()]
-    passed_fall = FreshmanEvalData.query.filter(
-        FreshmanEvalData.freshman_eval_result == "Passed"
-    ).distinct()
-
-    for intro_member in passed_fall:
-        voting_list.append(intro_member.uid)
-
-    for active_member in active_members:
-        if active_member not in intro_members:
-            voting_list.append(active_member)
-
-    return voting_list
-
-
 @dashboard_bp.route('/dashboard/')
 def display_dashboard():
     log = logger.new(user_name=request.headers.get("x-webauth-user"),
                      request_id=str(uuid.uuid4()))
     log.info('frontend', action='display dashboard')
 
-    # get user data
-
-    user_name = request.headers.get('x-webauth-user')
+    # Get username from headers.
+    username = request.headers.get('x-webauth-user')
 
+    # Get the list of voting members.
     can_vote = get_voting_members()
+
+    member = ldap_get_member(username)
     data = dict()
-    data['username'] = user_name
-    data['name'] = ldap_get_name(user_name)
-    # Member Status
-    data['active'] = ldap_is_active(user_name)
-    # On-Floor Status
-    data['onfloor'] = ldap_is_onfloor(user_name)
-    # Voting Status
-    data['voting'] = bool(user_name in can_vote)
+    data['username'] = member.uid
+    data['name'] = member.cn
+    data['active'] = ldap_is_active(member)
+    data['onfloor'] = ldap_is_onfloor(member)
+    data['voting'] = bool(member.uid in can_vote)
 
     active_list = get_members_info()[0]
     data['voting_count'] = {"Voting Members": len(can_vote),
                             "Active Members": len(active_list)}
     # freshman shit
-    if ldap_is_intromember(user_name):
-        data['freshman'] = get_freshman_data(user_name)
+    if ldap_is_intromember(member):
+        data['freshman'] = get_freshman_data(username)
```

Comment:
In my opinion, we should do a `ldap_get_member` lookup based on whatever criteria (in this case, UID from Webauth) like you're doing above, then use that member object's attributes instead of continuing to use the header value. This will also aid in the transition away from Webauth.